doc/v5.0/man1/oshmem_info.1
.\" Man page contributed by Dirk Eddelbuettel <edd@debian.org>
.\" and released under the BSD license
.\" Copyright (c) 2008      Sun Microsystems, Inc.  All rights reserved.
.TH OSHMEM_INFO 1 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
oshmem_info - Display information about the Open MPI installation
.
.\" **************************
.\"    Synopsis Section
.\" **************************
.SH SYNOPSIS
.B oshmem_info [options]
.
.\" **************************
.\"    Description Section
.\" **************************
.SH DESCRIPTION
.PP
.B oshmem_info
provides detailed information about the Open MPI/SHMEM installation.  It can
be useful for at least three common scenarios:
.PP
1. Checking local configuration and seeing how Open MPI/SHMEM was installed.
.PP
2. Submitting bug reports / help requests to the Open MPI/SHMEM community
(see
.IR https://www.open-mpi.org/community/help/ )
.PP
3. Seeing a list of installed Open MPI/SHMEM plugins and querying what
MCA parameters they support.
.
.\" **************************
.\"    Options Section
.\" **************************
.SH OPTIONS
.B oshmem_info
accepts the following options:
.TP 8
.I \-a|\-\-all
Show all configuration options and MCA parameters
.TP 8
.I \-\-arch
Show architecture Open MPI/SHMEM was compiled on
.TP 8
.I \-c|\-\-config
Show configuration options
.TP 8
.I \-gmca|\-\-gmca <param> <value>
Pass global MCA parameters that are applicable to all contexts.
.TP 8
.I \-h|\-\-help
Shows help / usage message
.TP 8
.I \-\-hostname
Show the hostname that Open MPI/SHMEM was configured and built on
.TP 8
.I \-\-internal
Show internal MCA parameters (not meant to be modified by users)
.TP 8
.I \-\-level
Show only variables with at most this level (1-9). The default is 1 unless
\-\-all is specified without \-\-level in which case the default is 9. See
the LEVEL section for more information.
.TP 8
.I \-mca|\-\-mca <param> <value>
Pass context-specific MCA parameters; they are considered global if --gmca is
not used and only one context is specified.
.TP 8
.I \-\-param <type> <component>
Show MCA parameters.  The first parameter is the type of the component
to display; the second parameter is the specific component to display
(or the keyword "all", meaning "display all components of this type").
.TP 8
.I \-\-parsable
When used in conjunction with other parameters, the output is
displayed in a machine-parsable format
.I \-\-parseable
Synonym for --parsable
.TP 8
.I \-\-path <type>
Show paths that Open MPI/SHMEM was configured with.  Accepts the following
parameters: prefix, bindir, libdir, incdir, pkglibdir, sysconfdir.
.TP 8
.I \-\-pretty
When used in conjunction with other parameters, the output is
displayed in 'prettyprint' format (default)
.TP 8
.I \-\-selected-only
Show only variables from selected components.
.TP 8
.I \-V|\-\-version
Show version of Open MPI/SHMEM.
.
.\" **************************
.\"    Levels Section
.\" **************************
.SH LEVELS
.TP 4
1
Basic information of interest to users
.
.TP
2
Detailed information of interest to users
.
.TP
3
All remaining information of interest to users
.
.TP
4
Basic information required for tuning
.
.TP
5
Detailed information required for tuning
.
.TP
6
All remaining information required for tuning
.
.TP
7
Basic information for MPI implementors
.
.TP
8
Detailed information for MPI implementors
.
.TP
9
All remaining information for MPI implementors
.
.\" **************************
.\"    Examples Section
.\" **************************
.SH EXAMPLES
.TP 4
oshmem_info
Show the default output of options and listing of installed
components in a human-readable / prettyprint format.
.
.
.TP
oshmem_info --parsable
Show the default output of options and listing of installed
components in a machine-parsable format.
.
.
.TP
oshmem_info --param btl openib
Show the MCA parameters of the "openib" BTL component in a
human-readable / prettyprint format.
.
.
.TP
oshmem_info --param btl openib --parsable
Show the MCA parameters of the "openib" BTL component in a
machine-parsable format.
.
.
.TP
oshmem_info --path bindir
Show the "bindir" that Open MPI was configured with.
.
.
.TP
oshmem_info --version ompi full --parsable
Show the full version numbers of Open MPI/SHMEM (including the ORTE and OPAL
version numbers) in a machine-readable format.
.
.
.TP
oshmem_info --version btl major
Show the major version number of all BTL components in a prettyprint
format.
.
.
.TP
oshmem_info --version btl:tcp minor
Show the minor version number of the TCP BTL component in a
prettyprint format.
.
.
.TP
oshmem_info --all
Show
.I all
information about the Open MPI/SHMEM installation, including all components
that can be found, the MCA parameters that they support, versions of
Open MPI/SHMEM and the components, etc.
.
.
.\" **************************
.\"    Authors Section
.\" **************************
.SH AUTHORS
The Open MPI/SHMEM maintainers -- see
.I https://www.open-mpi.org/
or the file
.IR AUTHORS .
.PP
This manual page was originally contributed by Dirk Eddelbuettel
<edd@debian.org>, one of the Debian GNU/Linux maintainers for Open
MPI, and may be used by others.


doc/v3.1/man1/oshmem_info.1.php
<?php
$topdir = "../../..";
$title = "oshmem_info(1) man page (version 3.1.6)";
$meta_desc = "Open MPI v3.1.6 man page: oshmem_info(1)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
   <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
oshmem_info - Display information about the Open MPI installation

<h2><a name='sect1' href='#toc1'>Synopsis</a></h2>
<b>oshmem_info [options]</b>
<h2><a name='sect2' href='#toc2'>Description</a></h2>
<p>
<b>oshmem_info</b> provides detailed
information about the Open MPI/SHMEM installation.  It can be useful for
at least three common scenarios: <p>
1. Checking local configuration and seeing
how Open MPI/SHMEM was installed. <p>
2. Submitting bug reports / help requests
to the Open MPI/SHMEM community (see <i>http://www.open-mpi.org/community/help/</i>)
<p>
3. Seeing a list of installed Open MPI/SHMEM plugins and querying what MCA
parameters they support.
<h2><a name='sect3' href='#toc3'>Options</a></h2>
<b>oshmem_info</b> accepts the following options:

<dl>

<dt><i>-a|--all</i> </dt>
<dd>Show all configuration options and MCA parameters </dd>

<dt><i>--arch</i> </dt>
<dd>Show architecture
Open MPI/SHMEM was compiled on </dd>

<dt><i>-c|--config</i> </dt>
<dd>Show configuration options </dd>

<dt><i>-gmca|--gmca</i>
&lt;param&gt; &lt;value&gt; </dt>
<dd>Pass global MCA parameters that are applicable to all contexts.
</dd>

<dt><i>-h|--help</i> </dt>
<dd>Shows help / usage message </dd>

<dt><i>--hostname</i> </dt>
<dd>Show the hostname that Open MPI/SHMEM
was configured and built on </dd>

<dt><i>--internal</i> </dt>
<dd>Show internal MCA parameters (not
meant to be modified by users) </dd>

<dt><i>--level</i> </dt>
<dd>Show only variables with at most this
level (1-9). The default is 1 unless --all is specified without --level in which
case the default is 9. See the LEVEL section for more information. </dd>

<dt><i>-mca|--mca</i>
&lt;param&gt; &lt;value&gt; </dt>
<dd>Pass context-specific MCA parameters; they are considered global
if --gmca is not used and only one context is specified. </dd>

<dt><i>--param</i> &lt;type&gt; &lt;component&gt;
</dt>
<dd>Show MCA parameters.  The first parameter is the type of the component to
display; the second parameter is the specific component to display (or
the keyword "all", meaning "display all components of this type"). </dd>

<dt><i>--parsable</i>
</dt>
<dd>When used in conjunction with other parameters, the output is displayed
in a machine-parsable format <i>--parseable</i> Synonym for --parsable </dd>

<dt><i>--path</i> &lt;type&gt; </dt>
<dd>Show
paths that Open MPI/SHMEM was configured with.  Accepts the following parameters:
prefix, bindir, libdir, incdir, pkglibdir, sysconfdir. </dd>

<dt><i>--pretty</i> </dt>
<dd>When used
in conjunction with other parameters, the output is displayed in &rsquo;prettyprint&rsquo;
format (default) </dd>

<dt><i>--selected-only</i> </dt>
<dd>Show only variables from selected components.
</dd>

<dt><i>-V|--version</i> </dt>
<dd>Show version of Open MPI/SHMEM.     </dd>
</dl>

<h2><a name='sect4' href='#toc4'>Levels</a></h2>

<ol>
<li>Basic information of
interest to users  </li><li>Detailed information of interest to users  </li><li>All remaining
information of interest to users  </li><li>Basic information required for tuning
 </li><li>Detailed information required for tuning  </li><li>All remaining information required
for tuning  </li><li>Basic information for MPI implementors  </li><li>Detailed information
for MPI implementors  </li><li>All remaining information for MPI implementors
  </li>
</ol>

<h2><a name='sect5' href='#toc5'>Examples</a></h2>

<dl>

<dt>oshmem_info </dt>
<dd>Show the default output of options and listing of
installed components in a human-readable / prettyprint format.   </dd>

<dt>oshmem_info
--parsable </dt>
<dd>Show the default output of options and listing of installed components
in a machine-parsable format.   </dd>

<dt>oshmem_info --param btl openib </dt>
<dd>Show the MCA
parameters of the "openib" BTL component in a human-readable / prettyprint
format.   </dd>

<dt>oshmem_info --param btl openib --parsable </dt>
<dd>Show the MCA parameters
of the "openib" BTL component in a machine-parsable format.   </dd>

<dt>oshmem_info
--path bindir </dt>
<dd>Show the "bindir" that Open MPI was configured with.   </dd>

<dt>oshmem_info
--version ompi full --parsable </dt>
<dd>Show the full version numbers of Open MPI/SHMEM
(including the ORTE and OPAL version numbers) in a machine-readable format.
  </dd>

<dt>oshmem_info --version btl major </dt>
<dd>Show the major version number of all BTL
components in a prettyprint format.   </dd>

<dt>oshmem_info --version btl:tcp minor
</dt>
<dd>Show the minor version number of the TCP BTL component in a prettyprint
format.   </dd>

<dt>oshmem_info --all </dt>
<dd>Show <i>all</i> information about the Open MPI/SHMEM
installation, including all components that can be found, the MCA parameters
that they support, versions of Open MPI/SHMEM and the components, etc.
    </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Authors</a></h2>
The Open MPI/SHMEM maintainers -- see <i><i>http://www.openmpi.org/</i></i> or
the file <i>AUTHORS</i>. <p>
This manual page was originally contributed by Dirk Eddelbuettel
&lt;email-address-removed&gt;, one of the Debian GNU/Linux maintainers for Open MPI, and
may be used by others. <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Synopsis</a></li>
<li><a name='toc2' href='#sect2'>Description</a></li>
<li><a name='toc3' href='#sect3'>Options</a></li>
<li><a name='toc4' href='#sect4'>Levels</a></li>
<li><a name='toc5' href='#sect5'>Examples</a></li>
<li><a name='toc6' href='#sect6'>Authors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
